token/cli/src/main.rs

```diff
@@ -8456,6 +9003,87 @@ mod tests {
             .unwrap();
         assert!(bool::from(extension.allow_non_confidential_credits));
 
+        // deposit confidential tokens
+        let deposit_amount = 100.0;
+        mint_tokens(&config, &payer, token_pubkey, deposit_amount, token_account).await;
+
+        process_test_command(
+            &config,
+            &payer,
+            &[
+                "spl-token",
+                CommandName::DepositConfidentialTokens.into(),
+                &token_pubkey.to_string(),
+                &deposit_amount.to_string(),
+            ],
+        )
+        .await
+        .unwrap();
+
+        // apply pending balance
+        process_test_command(
+            &config,
+            &payer,
+            &[
+                "spl-token",
+                CommandName::ApplyPendingBalance.into(),
+                &token_pubkey.to_string(),
+            ],
+        )
+        .await
+        .unwrap();
+
+        // confidential transfer
+        let destination_account = create_auxiliary_account(&config, &payer, token_pubkey).await;
+        process_test_command(
+            &config,
+            &payer,
+            &[
+                "spl-token",
+                CommandName::ConfigureConfidentialTransferAccount.into(),
+                "--address",
+                &destination_account.to_string(),
+            ],
+        )
+        .await
+        .unwrap(); // configure destination account for confidential transfers first
+
+        // NOTE: the test fails due to transaction size limit
```

Comment:
The next step is to add support for split proofs for this to pass, right?

Reply:
Yep that would be the next step. Since this PR is quite long already, should we land this PR once the existing issues are addressed and then add the split proofs in a follow-up?

Reply:
Yeah that's fine, it'll require #5663 anyway!